src/io/jesi/backpack/spy.cljc

```diff
@@ -25,19 +25,19 @@
     (name form)
     (str form)))
 
-(defn- line-number [form]
-  (let [f (or (when (and *file*
-                         (not= "NO_SOURCE_PATH" *file*)
-                         (not= \/ (nth *file* 0)))
-                *file*)
+(defn- line-number [file form]
+  (let [f (or (when (and file
+                         (not= "NO_SOURCE_PATH" file)
+                         (not= \/ (nth file 0)))
+                file)
               *ns*)
         line (:line (meta form))]
     (str f \: line)))
 
 (defmacro prn [& more]
   `(when-debug
      (when *enabled*
-       (println ~@(let [line (line-number &form)]
+       (println ~@(let [line (line-number *file* &form)]
```

Comment:
*file* is clojure only. By passing it to the line-number fn we remove the warnings